core/types/amount.go

```diff
@@ -83,43 +185,63 @@ func NewAmount(f float64) (Amount, error) {
 	case math.IsInf(f, 1):
 		fallthrough
 	case math.IsInf(f, -1):
-		return 0, errors.New("invalid coin amount")
+		return &Amount{0, MEERID}, errors.New("invalid coin amount")
+	}
+
+	return &Amount{round(f * AtomsPerCoin), MEERID}, nil
+}
+
+func NewMeer(a uint64) (*Amount, error) {
+	amt := Amount{int64(a), MEERID}
```

Comment:
why don't just define Amount.Value as uint64 to avoid conversion

Reply:
because I don't want to break the original type definition, if change int to uint which required change & double check all exist code.

Reply:
for the original definitions, only types.Amount is int64, but the others are uint64, so which one should be more reasonable?

types.Amount
https://github.com/Qitmeer/qitmeer/blob/741b333219c9fd3a6bf063c23371a3373ccf7e22/core/types/amount.go#L56

others
https://github.com/Qitmeer/qitmeer/blob/741b333219c9fd3a6bf063c23371a3373ccf7e22/core/blockchain/stxo.go#L85
https://github.com/Qitmeer/qitmeer/blob/741b333219c9fd3a6bf063c23371a3373ccf7e22/core/blockchain/utxo.go#L46
https://github.com/Qitmeer/qitmeer/blob/741b333219c9fd3a6bf063c23371a3373ccf7e22/core/json/tx.go#L75
https://github.com/Qitmeer/qitmeer/blob/741b333219c9fd3a6bf063c23371a3373ccf7e22/core/types/tx.go#L1005
https://github.com/Qitmeer/qitmeer/blob/741b333219c9fd3a6bf063c23371a3373ccf7e22/ledger/ledger.go#L15
https://github.com/Qitmeer/qitmeer/blob/741b333219c9fd3a6bf063c23371a3373ccf7e22/ledger/ledger.go#L20

Reply:
using signed type might be better for comparing and calculation. such as `a - b < 0`

Reply:
ok